logdevice/common/test/NodesConfigurationPollerTest.cpp

```diff
@@ -0,0 +1,116 @@
+#include "logdevice/common/test/TestUtil.h"
+#include "logdevice/common/toString.h"
+
+using namespace facebook::logdevice;
+
+namespace {
+
+class NodesConfigurationPollerExtended : public NodesConfigurationPoller {
+ public:
+  static std::vector<node_index_t> nodesCalled;
+  virtual ~NodesConfigurationPollerExtended() {}
+
+  NodesConfigurationPollerExtended(
+      Poller::Options options,
+      VersionExtFn version_fn,
+      Callback cb,
+      folly::Optional<u_int32_t> node_order_seed,
+      folly::Optional<Version> conditional_base_version)
+      : NodesConfigurationPoller(std::move(options),
+                                 std::move(version_fn),
+                                 std::move(cb),
+                                 node_order_seed,
+                                 std::move(conditional_base_version)) {}
+
+ protected:
+  Poller::RequestResult sendRequestToNode(Poller::RoundID round,
+                                          node_index_t node) {
+    /* nodesCalled.push_back(node); */
+    return Poller::RequestResult::OK;
+  }
+
+  ClusterState* getClusterState() {
+    return nullptr;
+  }
+
+  std::shared_ptr<const configuration::nodes::NodesConfiguration>
+  getNodesConfiguration() const {
+    auto node_config = createSimpleNodesConfig(1 /* 1 node */);
+    auto server_config = ServerConfig::fromDataTest(
+        "random_node_selector_test", std::move(node_config));
+
+    auto nodes = server_config->getNodes();
+    auto node_id =
+        NodeID(nodes.begin()->first, nodes.begin()->second.generation);
+    return server_config->getNodesConfigurationFromServerConfigSource();
+  }
+
+  folly::Optional<node_index_t> getMyNodeID() const {
+    return folly::none;
+  }
+  // return true if the NCPoller is running in the bootstrapping environment
+  // (e.g., NodesConfigurationInit). If so, conditional polling will be disabled
+  bool isBootstrapping() const {
+    return false;
+  }
+
+ public:
+  auto testFunction() {
+    return *selectionFunction({1, 2, 3, 4, 5}, {}, {}, {}, 1, 0).begin();
+  }
+};
+
+std::vector<node_index_t> NodesConfigurationPollerExtended::nodesCalled;
+
+TEST(NodConfigurationPoller, simple) {
+  std::vector<node_index_t> resultsAcrossInvocations;
+
+  for (int i = 0; i < 10; i++) {
+    NodesConfigurationPoller::Poller::Options options;
+    options.mode = NodesConfigurationPoller::Poller::Mode::ONE_TIME;
+    auto nodesConfigurationPollerExtended = NodesConfigurationPollerExtended(
+        options,
+        configuration::nodes::NodesConfigurationCodec::extractConfigVersion,
+        [this](Status st,
+               NodesConfigurationPoller::Poller::RoundID /*round*/,
+               folly::Optional<std::string> config_str) {},
+        folly::Random::rand32(),
+        folly::none);
+
+    // checking that using the same object returns the same node everytime
+    auto result = nodesConfigurationPollerExtended.testFunction();
```

Comment:
I don't think this needs to be a test function. I think it's easier to read if it's inlined here.

Reply:
since the scope of selectionFunction is private, I exposed it via a public function.
Is there any other way to directly call the selectionFunction?